### PR TITLE
Add e2e test job to odh operator v2.0

### PR DESCRIPTION
**Pull request by VaishnaviHire**

### CODE REVIEW

File: ci-operator/config/opendatahub-io/opendatahub-operator/opendatahub-io-opendatahub-operator-feature-rearchitecture.yaml [hunk 1]
```diff
@@ -54,6 +54,27 @@ tests:
...
+      OO_CHANNEL: fast
+      OO_INSTALL_NAMESPACE: openshift-operators
+      OO_PACKAGE: opendatahub-operator
+      OO_TARGET_NAMESPACES: '!all'
+    test:
+    - as: e2e
+      cli: latest
+      commands: |
+        unset GOFLAGS
+        make e2e-test -e OPERATOR_NAMESPACE=openshift-operators
```
Comment (zdtsw) [inline ci-operator/config/opendatahub-io/opendatahub-operator/opendatahub-io-opendatahub-operator-feature-rearchitecture.yaml:71]: get me wondered why in makefile, we set `openshift-operators-system` as fallback value
if we change to use `openshift-operators` then we do not need to pass variable here.

**Review by zdtsw — APPROVED** [review]
/lgtm

File: ci-operator/config/opendatahub-io/opendatahub-operator/opendatahub-io-opendatahub-operator-feature-rearchitecture.yaml (hunk 1 above)
Comment (VaishnaviHire) [reply ci-operator/config/opendatahub-io/opendatahub-operator/opendatahub-io-opendatahub-operator-feature-rearchitecture.yaml:71]: Its `opendatahub-operator-system` which is also created by default by sdk. I think we do not want local dev deployments to deploy in `openshift-operators` namespace always.

**Comment by VaishnaviHire** [comment]
expected failure, since make target is not merged into the branch

**Comment by VaishnaviHire** [comment]
/pj-rehearse ack

Tests passed successfully

**Pull request merged**